### PR TITLE
Extract path typing functionality from PopulateReturnTemplate into GetReturnTemplatePathTypes

### PR DESCRIPTION
**Pull request by faissalMT**

### CODE REVIEW

File: lib/local_authority/use_case/get_return_template_path_types.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+  def get_path_types_from_array_last_dependency(path, schema)
+    schema.dig(:items, :dependencies).values.each do |value|
+      path_types = get_state_path(value, path)
+      return [:array] + path_types if path_found?(path_types)
+    end
+  end
+
+  def array_has_item_dependencies?(schema)
+    Common::HashUtils.hash_has_path(schema, %i[items dependencies])
```
Comment (danielburnley) [inline lib/local_authority/use_case/get_return_template_path_types.rb]: You could also do this with a refinement (https://github.com/homes-england/monitor-api/blob/e3f251851de6093e0d5b369d35e1d9b0242769b6/lib/local_authority/refinement/deep_copy.rb), not necessary but worth looking at

File: lib/local_authority/use_case/get_return_template_path_types.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    [:unsupported_type]
+  end
+
+  def get_schema_types_from_array(path, schema)
+    acquired_path = [:not_found]
+    if array_has_property?(path, schema)
+      acquired_path = get_path_types_from_array_properties(path, schema)
+    end
+    if array_has_item_dependencies?(schema) && path_not_found?(acquired_path)
+      acquired_path = get_path_types_from_array_last_dependency(path, schema)
```
Comment (danielburnley) [inline lib/local_authority/use_case/get_return_template_path_types.rb]: > from array last dependency

Should that be `last dependency` still?

File: lib/local_authority/use_case/get_return_template_path_types.rb (hunk 1 above)
Comment (faissalMT) [reply lib/local_authority/use_case/get_return_template_path_types.rb]: Sounds good to me

File: lib/local_authority/use_case/get_return_template_path_types.rb (hunk 2 above)
Comment (faissalMT) [reply lib/local_authority/use_case/get_return_template_path_types.rb]: No

**Pull request merged**